src/form.js

```diff
@@ -983,6 +983,21 @@ define([
             var _this = this;
             this._logicManager.forEachBrokenReference(updateReferences);
         },
+        isReferencedByOtherMugs: function (mug, except) {
+            var ref = false,
+                ufids = {},
+                exclude = _.object(_.map(except, function (mug) {
+                    return [mug.ufid, null];
+                }));
+            exclude[mug.ufid] = null;
+            ufids[mug.ufid] = null;
+            this._logicManager.forEachReferencingProperty(ufids, function (mug) {
+                if (!exclude.hasOwnProperty(mug.ufid)) {
+                    ref = true;
```

Comment:
Agreed, although it's notoriously hard to break out of `_.each`. I'll see if I can redesign it to allow that.